app/src/Homepage/Homepage.tsx

```diff
@@ -85,10 +85,48 @@ export const LevelTab: React.FC<{ classlevel: string }> = ({ classlevel }) => {
     );
 }
 
-export const InnerPage: React.FC = () => {
+export const HomePage: React.FC = () => {
     return (
-        <div className="innerpage">
+        <div className="homepage">
             {/* This is the content of the inner page */}
         </div>
     );
 }
+
+interface ClassListProps {
+    classLevelNumber: string;
+}
+  
+export const ClassList: React.FC<ClassListProps> = ({ classLevelNumber }) => {
+    const [classList, setClassList] = useState<any[]>([]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+        try {
+            const response = await fetch(`/api/GetClassData?level=${classLevelNumber}`);
+            const data = await response.json();
+            setClassList(data);
+        } catch (error) {
+            console.error('Error fetching data:', error);
+        }
+        };
+
+        fetchData();
+    }, [classLevelNumber]);
+
+    return (
+        <div className="innerpage">
+          <div className="class-list">
+            {classList.map((classItem) => (
+              <div key={classItem.id} className="card">
```

Comment:
May be mistaken, but is the field name a class' id "class_id" not "id"? Key probably not that significant though.